### ISSUE TITLE
Button links in global Pull requests / Issues pages are messed up

### ISSUE THREAD

**Issue by mcornella** [sindresorhus/refined-github#1919]
Links to reproduce: [Pull requests](https://github.com/pulls), [Issues](https://github.com/issues). Then click either the `Commented` or `Yours` button. 

You can see in the screenshot that the href attributes of the original buttons (`Created`, `Assigned`, `Mentioned` and `Review requests` in the Pull requests page) don't have a space (`+`) after the `archived:false` attribute.

![image](https://user-images.githubusercontent.com/1441704/55908695-6959c600-5bda-11e9-853b-b4f9bf1e75a9.png)

This is what it loads if I press one of them:

![image](https://user-images.githubusercontent.com/1441704/55908764-8f7f6600-5bda-11e9-85a4-c64b9316ef67.png)

And this is how it looks with the extension disabled:

![image](https://user-images.githubusercontent.com/1441704/55908810-a9b94400-5bda-11e9-8cc7-324bf8a9acac.png)

I have version 19.4.7.1857 of the extension.

Great work on the project. Keep it up! :)

**Comment by mcornella** [comment]
This was probably introduced in #1883 because if I disable the `global-discussion-list-filters` the buttons aren't there anymore, and the links are correct.

**Comment by bfred-it** [comment]
Indeed, this is due to: `const queryRegex = new RegExp(`(^|\\s)?${query}(\\s|$)?`);`, which drops both surrounding spaces when it’s later used in a `.replace`

**Comment by mcornella** [comment]
That's right, droping one side of the regexp fixes it. There is also an issue in how the `is:issues` parameter is added for the Issues page (it should be `is:issue` singular). Here's a possible patch:
```diff
diff --git a/source/features/global-discussion-list-filters.tsx b/source/features/global-discussion-list-filters.tsx
index 9147821..7cf3145 100644
--- a/source/features/global-discussion-list-filters.tsx
+++ b/source/features/global-discussion-list-filters.tsx
@@ -5,7 +5,7 @@ import {getUsername} from '../libs/utils';

 function init() {
     const defaultQuery = 'is:open archived:false ';
-    const type = location.pathname === '/issues' ? ' is:issues' : ''; // Without this, the Issues page also displays PRs
+    const type = location.pathname === '/issues' ? ' is:issue' : ''; // Without this, the Issues page also displays PRs
     const links = [
         ['Commented', `commenter:${getUsername()}`],
         ['Yours', `user:${getUsername()}`]
@@ -17,7 +17,7 @@ function init() {
         const link = <a href={`?${url}`} class="subnav-item">{label}</a>;

         // Create regex for current query, including possible spaces around it, to drop it in .replace()
-        const queryRegex = new RegExp(`(^|\\s)?${query}(\\s|$)?`);
+        const queryRegex = new RegExp(`${query}(\\s|$)?`);
         const isCurrentPage = queryRegex.test(new URLSearchParams(location.search).get('q'));

         // Highlight it, if that's the current page
```

**Comment by bfred-it** [comment]
I was gonna say that “the space is there for a reason and it can’t be dropped” but there’s no *valid* situation where the leading space matters (because `wowcommenter:bfred-it` isn’t a valid query)

So you can open a PR with that

**Comment by mcornella** [comment]
We could also substitute it with a word boundary character (`\b`) if you're concerned about this. I'll play with it a little bit more and submit a PR once it's ready.

**Comment by bfred-it** [comment]
That’s how it was at some point but that also matches `user:bfred` in `user:bfred-it` when used after

**Comment by mcornella** [comment]
> That’s how it was at some point but that also matches `user:bfred` in `user:bfred-it` when used after. The space is the right separator to use, but it shouldn’t _also_ be dropped from _both_ sides.

This is also happening with the current code given that the separators around the query are optional in the regex, given that it's using `?`. I've put together a PR that fixes this as well: #1926.

I finally wound up using split + filter + join to remove the query parameter, and it seems to be working well. Thanks for the guidance!
